guidance/_grammar.py

```diff
@@ -243,6 +243,28 @@ def match_byte(self, byte):
     def max_tokens(self):
         return 1000000000000
 
+class Box(Terminal):
```

Comment:
Probably wants a different name. Could re-use the old `Placeholder` name?

Reply:
I'd only use `Placeholder` for something which is actually going to be _replaced_ (and not via that fun Python thing of resetting `self.__class__` which Guidance used to do). I'd say `Box` is a bit nondescript, though. `DeferredGrammar` perhaps? Or `DeferredReference`